tst/EnergyPlus/unit/GroundHeatExchangers.unit.cc

```diff
@@ -1333,6 +1336,360 @@ TEST_F(EnergyPlusFixture, GroundHeatExchangerTest_System_calcGFunction_Check)
+                          "    Only Water Loop Operation,  !- Name",
+                          "    PlantEquipmentOperation:CoolingLoad,  !- Control Scheme 1 Object Type",
+                          "    Only Water Loop Cool Operation All Hours,  !- Control Scheme 1 Name",
+                          "    HVACTemplate-Always 1;   !- Control Scheme 1 Schedule Name",
+
+                          "SetpointManager:Scheduled:DualSetpoint,",
+                          "    Ground Loop Temp Manager,!- Name",
+                          "    Temperature,             !- Control Variable",
+                          "    HVACTemplate-Always 34,  !- High Setpoint Schedule Name",
+                          "    HVACTemplate-Always 20,  !- Low Setpoint Schedule Name",
+                          "    Ground Loop Supply Setpoint Nodes;  !- Setpoint Node or NodeList Name",
+
+                          "Schedule:Compact,",
+                          "    HVACTemplate-Always 4,   !- Name",
+                          "    HVACTemplate Any Number, !- Schedule Type Limits Name",
+                          "    Through: 12/31,          !- Field 1",
+                          "    For: AllDays,            !- Field 2",
+                          "    Until: 24:00,4;          !- Field 3",
+
+                          "Schedule:Compact,",
+                          "    HVACTemplate-Always 34,  !- Name",
+                          "    HVACTemplate Any Number, !- Schedule Type Limits Name",
+                          "    Through: 12/31,          !- Field 1",
+                          "    For: AllDays,            !- Field 2",
+                          "    Until: 24:00,34;         !- Field 3",
+
+                          "Schedule:Compact,",
+                          "    HVACTemplate-Always 20,  !- Name",
+                          "    HVACTemplate Any Number, !- Schedule Type Limits Name",
+                          "    Through: 12/31,          !- Field 1",
+                          "    For: AllDays,            !- Field 2",
+                          "    Until: 24:00,20;         !- Field 3"});
+
+    // Envr variable
+    state->dataSysVars->DisableGLHECaching = true;
+
+    // Setup
+    ASSERT_TRUE(process_idf(idf_objects));
+    ProcessScheduleInput(*state);
+    state->dataScheduleMgr->ScheduleInputProcessed = true;
+    GetPlantLoopData(*state);
+    GetPlantInput(*state);
+    SetupInitialPlantCallingOrder(*state);
+    SetupBranchControlTypes(*state);
+
+    auto &thisGLHE(state->dataGroundHeatExchanger->verticalGLHE[0]);
+    thisGLHE.loopNum = 1;
+    state->dataLoopNodes->Node(thisGLHE.inletNodeNum).Temp = 20;
+    thisGLHE.designFlow = 0.00075708;
+
+    Real64 rho = 998.207; // Density at 20 C using CoolProp
+    thisGLHE.designMassFlow = thisGLHE.designFlow * rho;
+
+    thisGLHE.myRespFactors->maxSimYears = 1;
+
+    thisGLHE.calcGFunctions(*state);
+
+    Real64 const tolerance = 0.1;
+
+    // Test g-function values from GLHEPro
```

Comment:
If this is truly testing the g-function values from GLHEPRO, we may have an issue if the field is large enough. For smaller fields this won't be a problem, but the UIFT approximation becomes quite a bit different (as much as +15%) when fields become larger. Have you made sure this wont break @mitchute ?